CHANGELOG_PENDING.md

```diff
@@ -1,5 +1,7 @@
 ### Improvements
 
+- [sdk] - Add `RetainOnDelete` as a resource option.
```

Comment:
Open to name suggestions if anyone thinks RetainOnDelete isn't clear enough. Another idea was "OrphanOnDelete".

Reply:
I prefer RetainOnDelete over OrphanOnDelete. 

Note this also aligns with CloudFormation’s similar feature: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-deletionpolicy.html. ARM and Terraform don’t have this feature. 

Were there other candidate options we should consider?